src/v/pandaproxy/rest/handlers.cc

```diff
@@ -165,6 +171,29 @@ get_topics_records(server::request_t rq, server::reply_t rp) {
       parse::query_param<std::chrono::milliseconds>(*rq.req, "timeout")};
     int32_t max_bytes{parse::query_param<int32_t>(*rq.req, "max_bytes")};
 
+    auto handler = [user{rq.user},
+                    authn_method{rq.authn_method},
+                    offset{offset},
+                    timeout{timeout},
+                    max_bytes{max_bytes},
+                    res_fmt,
+                    tp{std::move(tp)}](kafka_client_cache& cache) mutable {
+        client_ptr client = cache.fetch_or_insert(user, authn_method);
+        return client
+          ->fetch_partition(std::move(tp), offset, max_bytes, timeout)
+          .then([res_fmt](kafka::fetch_response res) {
+              ::json::StringBuffer str_buf;
+              ::json::Writer<::json::StringBuffer> w(str_buf);
+
+              ppj::rjson_serialize_fmt(res_fmt)(w, std::move(res));
+
+              // TODO Ben: Prevent this linearization
+              ss::sstring json_rslt = str_buf.GetString();
+              return ss::make_ready_future<ss::sstring>(json_rslt);
```

Comment:
`json_rslt` could be quite large, probably best to avoid the copy, perhaps:
```suggestion
              return ss::make_ready_future<ss::sstring>(str_buf.GetString());
```